tests/iter.rs

```diff
@@ -25,6 +25,16 @@ fn struct_iter() {
     assert_eq!(paths.next(), None);
 }
 
+#[test]
+fn struct_iter_indices() {
+    let mut paths = [([0, 0], 1), ([1, 0], 1), ([2, 0], 2)].into_iter();
+    for (have, expect) in Settings::iter_indices().zip(&mut paths) {
+        assert_eq!(have, expect);
+    }
+    // Ensure that all fields were iterated.
+    assert_eq!(paths.next(), None);
+}
+
```

Comment:
It would be good to have a test case for the special "single root option type" thats covered by the `Ok(0)` branch in the iter methods.

Reply:
https://github.com/quartiq/miniconf/blob/bd88587de0ca5fc92724122f5b91f4cbdf22f35d/tests/option.rs#L17